src/pactflow/client/utils.ts

```diff
@@ -99,3 +106,119 @@ export async function addOpenAPISpecToSchema(inputSchema: any) {
 
   return inputSchema;
 }
+
+/**
+ * Get OpenAPI matcher recommendations using sampling.
+ *
+ * @param openAPI The OpenAPI document to analyze.
+ * @param server The SmartBear MCP server instance.
+ * @returns A promise that resolves to the matcher recommendations.
+ * @throws Error if unable to parse recommendations.
+ */
+export async function getOADMatcherRecommendations(
+  openAPI: OpenAPI,
+  server: Server
+): Promise<MatcherRecommendations> {
+  const matcherResponse = await server.createMessage({
+    messages: [
+      {
+        role: "user",
+        content: {
+          type: "text",
+          text: OADMatcherPrompt.replace("{0}", JSON.stringify(openAPI)),
+        },
+      },
+    ],
+    maxTokens: 1000,
+  });
+
+  const regex = /```json([\s\S]*?)```/;
```

Comment:
Given that `\S` is the complement of `\s`, then `[\s\S]` will match any character. So it would be more straightforward to have `.*` instead.

```suggestion
  const regex = /```json(.*?)```/;
```

There's also the slight chance of having `json5`, `jsonc`: ```json[c5]?(.*?)```

and there's also the slight chance of different cases. So I would do a case-insensitive search/match.